### PR TITLE
`Ares`: Add a file reader for the security configuration file

### PR DESCRIPTION
**Pull request by MarkusPaulsen**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Ares
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
Ares 2.0 should be able to load its configuration via a configuration file and either create a folder of test cases out of it (pre-compile-mode) or load the configuration at runtime (post-compile-mode).

### Description
Ares 2.0 implements security configuration loading and security test case creation using the facade, builder, and abstract factory patterns. The Policy class serves as the facade annotation for simplified library access, facilitating the creation and execution of security test cases defined by a policy file at a specified path. The policy file is transformed into a SecurityPolicy structure by the SecurityPolicyReaderAndDirector, which subsequently creates the respective security test cases via the implementation of the SecurityTestCaseAbstractFactoryAndBuilder class. The current test cases can either be architectural tests or aspect oriented programming configurations.
ClassDiagram-2

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: src/main/java/de/tum/cit/ase/ares/api/Policy.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.ANNOTATION_TYPE;
+
+/**
+ * This annotation is used to specify the security policy
+ * that should be used for {@link de.tum.cit.ase.ares.api.jupiter.PublicTest}
+ * or {@link de.tum.cit.ase.ares.api.jupiter.HiddenTest}.
+ * This annotation is optional and specified in TODO.
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/Policy.java:19]: We have a TODO here, was this intended?

File: src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * @author Markus Paulsen
+ * @version 2.0.0
+ * @since 2.0.0
+ */
+public enum JavaSupportedArchitectureTestCase {
+    /**
+     * Architecture test case for the file system interaction.
+     */
+    FILESYSTEMINTERACTION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java:14]:
```suggestion
    FILESYSTEM_INTERACTION,
```

File: src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Architecture test case for the network connection.
+     */
+    NETWORKCONNECTION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java:18]:
```suggestion
    NETWORK_CONNECTION,
```

File: src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Architecture test case for the command execution.
+     */
+    COMMANDEXECUTION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java:22]:
```suggestion
    COMMAND_EXECUTION,
```

File: src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Architecture test case for the thread creation.
+     */
+    THREADCREATION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java:26]:
```suggestion
    THREAD_CREATION,
```

File: src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Architecture test case for the package import.
+     */
+    PACKAGEIMPORT
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/architectureTest/JavaSupportedArchitectureTestCase.java:30]:
```suggestion
    PACKAGE_IMPORT
```

File: src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * @author Markus Paulsen
+ * @version 2.0.0
+ * @since 2.0.0
+ */
+public enum JavaSupportedAspectConfiguration {
+    /**
+     * Aspect configuration for the file system interaction.
+     */
+    FILESYSTEMINTERACTION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java:14]:
```suggestion
    FILESYSTEM_INTERACTION,
```

File: src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Aspect configuration for the network connection.
+     */
+    NETWORKCONNECTION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java:18]:
```suggestion
    NETWORK_CONNECTION,
```

File: src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Aspect configuration for the command execution.
+     */
+    COMMANDEXECUTION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java:22]:
```suggestion
    COMMAND_EXECUTION,
```

File: src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Aspect configuration for the thread creation.
+     */
+    THREADCREATION,
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java:26]:
```suggestion
    THREAD_CREATION,
```

File: src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * Aspect configuration for the package import.
+     */
+    PACKAGEIMPORT
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/aspectConfiguration/JavaSupportedAspectConfiguration.java:30]:
```suggestion
    PACKAGE_IMPORT
```

File: src/main/java/de/tum/cit/ase/ares/api/jqwik/JqwikSecurityExtension.java [hunk 12]
```diff
@@ -28,6 +35,22 @@ public int aroundPropertyProximity() {
...
+				Policy policy = policyAnnotation.get();
+				Path policyPath = Path.of(policy.value());
+				if (!policyPath.toFile().exists()) {
+					throw new SecurityException("Policy file does not exist at: " + policyPath);
+				}
+				new SecurityPolicyReaderAndDirector(policyPath).runSecurityTestCases();
+			}
+
+		}
 		var configuration = ConfigurationUtils.generateConfiguration(testContext);
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/jqwik/JqwikSecurityExtension.java:54]:
```suggestion
```
Unused configuration file

File: src/main/java/de/tum/cit/ase/ares/api/jupiter/JupiterSecurityExtension.java [hunk 13]
```diff
@@ -1,38 +1,61 @@
...
+                Policy policy = policyAnnotation.get();
+                Path policyPath = Path.of(policy.value());
+                if (!policyPath.toFile().exists()) {
+                    throw new SecurityException("Policy file does not exist at: " + policyPath);
+                }
+                new SecurityPolicyReaderAndDirector(policyPath).runSecurityTestCases();
+            }
+
+        }
+        var configuration = ConfigurationUtils.generateConfiguration(testContext);
```
Comment (sarpsahinalp) [inline src/main/java/de/tum/cit/ase/ares/api/jupiter/JupiterSecurityExtension.java:41]:
```suggestion
```
Unused configuration

**Review by sarpsahinalp — CHANGES_REQUESTED** [review]
Some small remarks

**Review by sarpsahinalp — COMMENTED** [review]
To adhere to a standard package naming, also used by the legacy code we should match the following regex `^[a-z_]+(\.[a-z_][a-z0-9_]*)*$` in the packages architectureTest, aspectConfiguration, securityTests.

File: src/main/java/de/tum/cit/ase/ares/api/Policy.java (hunk 1 above)
Comment (MarkusPaulsen) [reply src/main/java/de/tum/cit/ase/ares/api/Policy.java:19]: This should be replaced by a link to the policy file specification as soon as we have such a link.

**Review by sarpsahinalp — APPROVED** [review]
Approve after changes

**Review by az108 — APPROVED** [review]
LGTM after changes

**Pull request merged**